controller/data/queries.go

```diff
@@ -124,6 +129,24 @@ const (
 	appListQuery = `
 SELECT app_id, name, meta, strategy, release_id, deploy_timeout, created_at, updated_at
 FROM apps WHERE deleted_at IS NULL ORDER BY created_at DESC`
+	appListPageQuery = `
+SELECT a.app_id, a.name, a.meta, a.strategy, a.release_id, a.deploy_timeout, a.created_at, a.updated_at
+FROM apps AS a
+LEFT OUTER JOIN (SELECT app_id, created_at FROM apps WHERE app_id = $1 LIMIT 1) AS b ON true
+WHERE a.deleted_at IS NULL
+AND CASE
+	WHEN array_length($2::text[], 1) > 0
+		THEN a.app_id::text = ANY($2::text[])
+	ELSE true
+END
+AND match_label_filters($3, a.meta)
+AND CASE WHEN b IS NULL THEN true
+ELSE
+	a.created_at <= b.created_at
```

Comment:
The reasoning behind using an id instead of a timestamp is in case there are multiple rows with the same `created_at` which could break pagination. It seems `BeforeID` has become a misnomer here though as the results include that row (it should probably be renamed to something like `CursorID`).

Reply:
I've renamed to `CursorID` and switched to using the `created_at` column directly (since using the ID and selecting out its `created_at` has exactly the same effect). The cursor uses microsecond precision, so it's doubtful there will be multiple apps with the same timestamp (and if there is and they happen to be at the end of a page, well then they'll also appear at the start of the next page, I think that's subtle enough to not need further engineering to work around).